source/rules/sync-compatibility.txt

```diff
@@ -0,0 +1,168 @@
+   * - :json-expansion:`%%true`, :json-expansion:`%%false`
+     - Yes
+     - Yes
+   
+   * - :json-expansion:`%%values`, :json-expansion:`%%environment`
+     - Yes
+     - Yes with an :ref:`important consideration <flex-sync-expansions-consideration>`
+   
+   * - :json-expansion:`%%request`
+     - No
+     - No
+   
+   * - :json-expansion:`%%user`
+     - Yes
+     - Yes with an :ref:`important consideration <flex-sync-expansions-consideration>`
+   
+   * - :json-expansion:`%%this`, :json-expansion:`%%prev`, :json-expansion:`%%root`,  :json-expansion:`%%prevRoot`
+     - No. These expansions refer to the document. App Services evaluates "apply when" expressions at session start, so there's no document to evaluate.
+     - No. These expansions might access non-queryable fields of the document, which is not possible.
+   
+   * - :json-expansion:`%%partition`
+     - No
+     - No
+   
+   * - :json-operator:`%stringToOid`, :json-operator:`%oidToString`
+     - Yes
+     - Yes
+   
+   * - :json-operator:`%function`
+     - Yes
+     - No. App Services expands the expansions at the start of the session, so the function would not operate on a per-document basis.
+   
+   * - :json-operator:`$exists`
+     - Yes
+     - Yes
+   
+   * - :json-operator:`$in`, :json-operator:`$nin`
+     - Yes
+     - Yes. However, note that you cannot currently have an array field as a queryable field on a document.
+   
+   * - :json-operator:`$eq`, :json-operator:`$ne`, :json-operator:`$gt`, :json-operator:`$gte`, :json-operator:`$lt`, :json-operator:`$lte`  
+     - Yes
+     - Yes
+
+.. _flex-sync-expansions-consideration:
+
+.. important::
+
+   App Services causes a :ref:`client reset <client-resets>` if anything about
+   the role changed since the previous session.
+
+   At the start of a session, App Services expands all expansions in the "apply
+   when", ``document_filters.read``, and ``document_filters.write`` expressions
+   and stores the result. This has the following implications:
+
+   - If the value changes during a session, App Services continues
+     to use the value as it was at the time of session start.
+
+   - On the next session, if the value is different from what it was at
+     the start of this session, App Services causes a client
+     reset.
+   
+   - You cannot use the :json-operator:`%function` operator in read and
+     write rules. Functions would not operate on a per-document basis.
+   
+   - You cannot store permissions information (such as "which document
+     IDs may this user access?") in the user object.
+
+
+.. flexible-sync-permission-changes:
+
+Permission Changes
+------------------
+
+If a user's permissions have changed since the last Sync session, Sync triggers
+a :ref:`client reset <client-resets>` and re-downloads all data with the new
+permissions applied.
+
+A user's permissions might change in the following situations:
+
+- You :ref:`updated the Sync configuration <alter-your-sync-configuration>` to
```

Comment:
I think this should probably be "updated the Data Source configuration" instead or something similar, since that's where rules for flexible sync will be read from moving forward